### PR TITLE
Bump azure-mgmt-containerinstance

### PR DESCRIPTION
**Pull request by pankajastro**
Related to: https://github.com/apache/airflow/issues/30199

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/microsoft/azure/operators/container_instances.py [hunk 1]
```diff
@@ -90,7 +89,6 @@ class AzureContainerInstancesOperator(BaseOperator):
     :param restart_policy: Restart policy for all containers within the container group.
         Possible values include: 'Always', 'OnFailure', 'Never'
     :param ip_address: The IP address type of the container group.
-    :param network_profile: The network profile information for a container group.
```
Comment (eladkal) [inline airflow/providers/microsoft/azure/operators/container_instances.py:93]: This is breaking change.
Please set major version on `provider.yaml` we will also need entry in changelog.rst (top of the file) with the things that were broken and how to mitigate them.

Comment (pankajastro) [reply airflow/providers/microsoft/azure/operators/container_instances.py:93]: yes, still try a few things, will update the provider.yaml and changelog.rst  if needed

**Comment by strictness** [comment]
Hi, I'm just adding some additional info for you guys that will be relevant once you come to updating the hooks (related issue #34749)

> The hook in container_instance.py should use the azure.identity package instead of azure.common.credentials. [Link to code](https://github.com/apache/airflow/blob/main/airflow/providers/microsoft/azure/hooks/container_instance.py#L25)
> 
> Microsoft posted deprecation warnings about using that package since early 2021 - [azure.common](https://learn.microsoft.com/en-us/python/api/azure-common/azure.common.credentials?view=azure-python-previous).
> 
> https://github.com/Azure/azure-sdk-for-python/issues/16908#issuecomment-785216591, where the solution is to start using azure.identity instead.

File: airflow/providers/microsoft/azure/hooks/container_instance.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 import warnings
 from functools import cached_property
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, cast
 
 from azure.common.client_factory import get_client_from_auth_file, get_client_from_json_dict
 from azure.common.credentials import ServicePrincipalCredentials
```
Comment (strictness) [inline airflow/providers/microsoft/azure/hooks/container_instance.py]: Package is deprecated and no longer working. Consider using `azure.identity` ([relevant info](https://learn.microsoft.com/en-us/python/api/azure-common/?view=azure-python-previous))

Related issue on [azure-sdk-for-python](https://github.com/Azure/azure-sdk-for-python/issues/16908#issuecomment-785216591)

File: tests/providers/microsoft/azure/hooks/test_azure_container_instance.py [hunk 3]
```diff
@@ -52,7 +52,10 @@ def setup_test_cases(self, create_mock_connection):
                 conn_type="azure_container_instances",
                 login="login",
                 password="key",
-                extra={"tenantId": "tenant_id", "subscriptionId": "subscription_id"},
+                extra={
+                    "tenantId": "63e85d06-62e4-11ee-8c99-0242ac120002",
```
Comment (Lee-W) [inline tests/providers/microsoft/azure/hooks/test_azure_container_instance.py:56]: Is it a valid id? or just some random one?

File: airflow/providers/microsoft/azure/CHANGELOG.rst [hunk 4]
```diff
@@ -27,6 +27,20 @@
...
+
+Breaking changes
+~~~~~~~~~~~~~~~~
+
+.. warning::
+   In this version of the provider, we have removed network_profile param from AzureContainerInstancesOperator
+
+* Remove ``network_profile`` param from ``AzureContainerInstancesOperator``
+* Remove deprecated ``extra__azure__tenantId`` from azure_container_instance connection extras
+* Remove deprecated ``extra__azure__subscriptionId`` from azure_container_instance connection extras
```
Comment (Lee-W) [inline airflow/providers/microsoft/azure/CHANGELOG.rst:50]: Should we mention the named tuple `Volume` has been removed as well?

File: tests/providers/microsoft/azure/hooks/test_azure_container_instance.py (hunk 3 above)
Comment (pankajastro) [reply tests/providers/microsoft/azure/hooks/test_azure_container_instance.py:56]: it is Random. I use https://www.uuidgenerator.net/ to generate it. The new sdk throws a value error if tenantId is not valid uuid

File: airflow/providers/microsoft/azure/CHANGELOG.rst (hunk 4 above)
Comment (pankajastro) [reply airflow/providers/microsoft/azure/CHANGELOG.rst:50]: I have reverted this PTAL

File: airflow/providers/microsoft/azure/hooks/container_instance.py (hunk 2 above)
Comment (pankajastro) [reply airflow/providers/microsoft/azure/hooks/container_instance.py]: Removed ServicePrincipalCredentials and added ClientSecretCredential from `azure.identity`

File: airflow/providers/microsoft/azure/operators/container_instances.py (hunk 1 above)
Comment (pankajastro) [reply airflow/providers/microsoft/azure/operators/container_instances.py:93]: yeah, updated provider.yaml and changelog

**Review by eladkal — APPROVED** [review]

**Pull request merged**